source/common/config/bootstrap_json.cc

```diff
@@ -24,7 +25,7 @@ void BootstrapJson::translateClusterManagerBootstrap(
     auto* cluster = bootstrap.mutable_static_resources()->mutable_clusters()->Add();
     Config::CdsJson::translateCluster(*json_sds->getObject("cluster"),
                                       absl::optional<envoy::api::v2::core::ConfigSource>(),
-                                      *cluster);
+                                      *cluster, stats_options);
```

Comment:
This is kind of an interesting maze to follow for those of us that don't know this part of the code deeply.  But with a bit of grepping I think you can or should be able to pull the stats options from Cluster, which knows about ClusterManager, which knows about a bunch of things. I'm not sure if you can exactly get to the OptionsImpl from any of those, but it given all the things passed into ClusterManagerImpl's ctor, I think it would not be unreasonable to also have it see the options.

source/common/upstream/cluster_manager_impl.h

WDYT?

Reply:
I think it would be reasonable for ClusterManager and ListenerManager and HttpConnectionManager (for routes) to know about some higher-level thing with access to a StatsOptions (such as ServerImpl). Right now I've embedded StatsOptions inside Scope, and Scope as a member of a Server::Configuration::FactoryContext, which is already present in a lot of these locations. I'm just not sure it would really save us that much complexity.